packages/app/src/vis-packs/nexus/utils.ts

```diff
@@ -25,15 +24,25 @@ import type {
   NumArrayDataset,
 } from '@h5web/shared';
 
+import type { AttrValuesStore } from '../../providers/context';
 import { getAttributeValue, hasAttribute } from '../../utils';
 import type { NxData, SilxStyle } from './models';
 
-export function isNxDataGroup(group: Group): boolean {
-  return getAttributeValue(group, 'NX_class') === 'NXdata';
+export function isNxDataGroup(
+  group: Group,
+  attrValuesStore: AttrValuesStore
+): boolean {
+  return (
+    hasAttribute(group, 'signal') &&
+    attrValuesStore.getSingle(group, 'NX_class') === 'NXdata'
```

Comment:
As mentioned:

- we no longer show an error if an `NXdata` group has no `signal` attribute (via the assertion below)
- we avoid the fetch of the `NX_class` if the entity has no `signal` attribute